factory/apiResolverFactory.go

```diff
@@ -20,145 +20,276 @@ import (
+	apiWorkingDir := filepath.Join(args.Configs.FlagsConfig.WorkingDir, core.TemporaryPath)
+	argsSCQuery := &scQueryServiceArgs{
+		generalConfig:       args.Configs.GeneralConfig,
+		coreComponents:      args.CoreComponents,
+		dataComponents:      args.DataComponents,
+		stateComponents:     args.StateComponents,
+		processComponents:   args.ProcessComponents,
+		gasScheduleNotifier: args.GasScheduleNotifier,
+		messageSigVerifier:  args.CryptoComponents.MessageSignVerifier(),
+		systemSCConfig:      args.Configs.SystemSCConfig,
+		workingDir:          apiWorkingDir,
+	}
 
-	argsBuiltIn := builtInFunctions.ArgsCreateBuiltInFunctionContainer{
-		GasSchedule:     gasScheduleNotifier,
-		MapDNSAddresses: make(map[string]struct{}),
-		Marshalizer:     marshalizer,
-		Accounts:        accnts,
+	scQueryService, vmFactory, vmContainer, err := createScQueryService(argsSCQuery)
+	if err != nil {
+		return nil, err
 	}
-	builtInFuncFactory, err := builtInFunctions.NewBuiltInFunctionsFactory(argsBuiltIn)
+
+	builtInFuncs, err := createBuiltinFuncs(
+		args.GasScheduleNotifier,
+		args.CoreComponents.InternalMarshalizer(),
+		args.StateComponents.AccountsAdapter(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	argsTxTypeHandler := coordinator.ArgNewTxTypeHandler{
+		PubkeyConverter:  args.CoreComponents.AddressPubKeyConverter(),
+		ShardCoordinator: args.ProcessComponents.ShardCoordinator(),
+		BuiltInFuncNames: builtInFuncs.Keys(),
+		ArgumentParser:   parsers.NewCallArgsParser(),
+	}
+	txTypeHandler, err := coordinator.NewTxTypeHandler(argsTxTypeHandler)
 	if err != nil {
 		return nil, err
 	}
-	builtInFuncs, err := builtInFuncFactory.CreateBuiltInFunctionContainer()
+
+	txCostHandler, err := transaction.NewTransactionCostEstimator(
+		txTypeHandler,
+		args.CoreComponents.EconomicsData(),
+		scQueryService,
+		args.GasScheduleNotifier,
+	)
 	if err != nil {
 		return nil, err
 	}
 
-	cacherCfg := storageFactory.GetCacherFromConfig(generalConfig.SmartContractDataPool)
-	smartContractsCache, err := storageUnit.NewCache(cacherCfg)
+	totaltakedHandlerArgs := &totalStakedAPI.ArgsTotalStakedValueHandler{
+		ShardID:                     args.ProcessComponents.ShardCoordinator().SelfId(),
+		RoundDurationInMilliseconds: args.CoreComponents.GenesisNodesSetup().GetRoundDuration(),
+		InternalMarshalizer:         args.CoreComponents.InternalMarshalizer(),
+		Accounts:                    args.StateComponents.AccountsAdapter(),
+	}
+
+	totalStakedValueHandler, err := totalStakedAPI.CreateTotalStakedValueHandler(totaltakedHandlerArgs)
 	if err != nil {
 		return nil, err
 	}
 
+	apiArgs := external.ApiResolverArgs{
+		ScQueryService:     scQueryService,
+		StatusMetrics:      args.CoreComponents.StatusHandlerUtils().Metrics(),
+		TxCostHandler:      txCostHandler,
+		VmFactory:          vmFactory,
+		VmContainer:        vmContainer,
+		StakedValueHandler: totalStakedValueHandler,
+	}
+
+	return external.NewNodeApiResolver(apiArgs)
+}
+
+func createScQueryService(
```

Comment:
Was this new code copied "by hand" after the merge? Not questioning the reason for this as this kind of merge is hard to do, just double checking.

Reply:
part of it manually applied merge as it was previously in a different file, and part of it refactor

Reply:
right